src/main/java/sg/ncl/MainController.java

```diff
@@ -2744,6 +2748,105 @@ public String adminTeamUsage(Model model,
+                              final RedirectAttributes redirectAttributes,
+                              HttpSession session) throws IOException {
+
+        if (!validateIfAdmin(session)) {
+            return NO_PERMISSION_PAGE;
+        }
+
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        ZonedDateTime now = ZonedDateTime.now();
+        if (start == null) {
+            ZonedDateTime startDate = now.with(firstDayOfMonth());
+            start = startDate.format(formatter);
+        }
+        if (end == null) {
+            ZonedDateTime endDate = now.with(lastDayOfMonth());
+            end = endDate.format(formatter);
+        }
+
+        HttpEntity<String> request = createHttpEntityHeaderOnly();
+
+        ResponseEntity responseEntity;
+        try {
+            responseEntity = restTemplate.exchange(properties.getEnergyStatistics("startDate=" + start, "endDate=" + end), HttpMethod.GET, request, String.class);
+        } catch (RestClientException e) {
+            log.warn("Error connecting to sio team service for energy usage: {}", e);
+            redirectAttributes.addFlashAttribute(MESSAGE, ERR_SERVER_OVERLOAD);
+            return REDIRECT_ENERGY_USAGE;
+        }
+
+        String responseBody = responseEntity.getBody().toString();
+        JSONArray jsonArray = new JSONArray(responseBody);
+
+        // handling exceptions from SIO
+        if (RestUtil.isError(responseEntity.getStatusCode())) {
+            MyErrorResource error = objectMapper.readValue(responseBody, MyErrorResource.class);
+            ExceptionState exceptionState = ExceptionState.parseExceptionState(error.getError());
+            switch (exceptionState) {
+                case START_DATE_AFTER_END_DATE_EXCEPTION:
+                    log.warn("Get energy usage : Start date after end date error");
+                    redirectAttributes.addFlashAttribute(MESSAGE, ERR_START_DATE_AFTER_END_DATE);
+                    return REDIRECT_ENERGY_USAGE;
+
+                default:
+                    log.warn("Get energy usage : sio or deterlab adapter connection error");
+                    redirectAttributes.addFlashAttribute(MESSAGE, ERR_SERVER_OVERLOAD);
+                    return REDIRECT_ENERGY_USAGE;
+            }
+        } else {
+            log.info("Get energy usage info : {}", responseBody);
+        }
+
+        DecimalFormat df2 = new DecimalFormat(".##");
+
+        double sumEnergy = 0.00;
+        List<String> listOfDate = new ArrayList<>();
+        List<Double> listOfEnergy = new ArrayList<>();
+        ZonedDateTime currentZonedDateTime = convertToZonedDateTime(start);
+        String currentDate = null;
+        for (int i = 0; i < jsonArray.length(); i++) {
+            sumEnergy  += jsonArray.getDouble(i);
+
+            // add into listOfDate to display graph
+            currentDate = currentZonedDateTime.format(formatter);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace this "switch" statement by "if" statements to increase readability. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1301)